### ISSUE TITLE
Accessibility:  I provided solutions to be implemented based on WAVE evaluation

### ISSUE THREAD

**Issue by Trailblazer007** [jupyterhub/outreachy#122]
### What page is this for?

https://ipython.org/project.html

### WAVE accessibility report

![Screenshot from 2022-10-22 14-41-36](https://user-images.githubusercontent.com/64600704/197347519-a812ed8b-3946-442d-a84c-640ee5d7d61c.png)
My name is Fatoye Oluwaseun and I evaluated the accessibility of the page - https://ipython.org, in concordance with the WCGA 2 standards.

The results obtained from the assessment after using the WAVE tool are as follows:

1.  ### Language missing or invalid
The language of the document is not identified or a lang attribute value is invalid.
Identifying the language of the page or page elements allows screen readers to read the content in the appropriate language. It also facilitates the automatic translation of content. Therefore the `lang="en"` should be present in the head of the `html` code.

2. ### Contrast errors
There are twenty-four (24) instances of very low contrast between the _text-color_ and the _background-color_. Adequate contrast of text is necessary for all users, especially users with low vision.
To fix this, the contrast between the foreground (text) color and the background color should be increased. Large text (larger than 18 point or 14 point bold) does not require as much contrast as smaller text. Text is present that has a contrast ratio less than 4.5:1, or large text (larger than 18 point or 14 point bold) has a contrast ratio less than 3:1. WCAG requires that page elements have both foreground AND background colors defined (or inherited) that provide sufficient contrast. 
Therefore, to fix this, When text is presented over a background image, the text must have a background color defined (typically in CSS) that provides adequate text contrast when the background image is disabled or unavailable.

3. ### Alerts

- *Justified text*
There are three (3) cases of justified text.
_Problem statement_: Large blocks of justified text can negatively impact readability due to varying word/letter spacing and 'rivers of white' that flow through the text.

_Possible solution_: Remove the full justification from the text.

- *Suspicious alternative text*
The alternative text is either insufficient or contains extraneous information. In this case, it contains insufficient information.
_Current code_: `<img class="logo" src="_static/IPy_header.png" alt="Logo">`

_Possible solution_: The alternative text should be more descriptive of the corresponding image. Screen readers and browser presentation inform the user that the object is an image, so the alternative text of "image of..." (and similar) should be avoided. If the image does not convey content or if the content is presented in nearby text (e.g., a caption), null/empty alternative text (alt="") is appropriate.

- *Image with title*
There are two (2) instances of images with title.
An image has a title attribute value but no alt value.
The title attribute value for images that lack an alt attribute value will be presented to screen reader users. However, providing image content in the alt attribute typically value provides better accessibility, and should be used in most cases. The title attribute will generate a mouse hover tooltip which more or may not be desired - this tooltip will not be presented to touch screen or keyboard users.

_Current code_: `<img src="_static/nbviewer-thm-3.png" title="IPython Notebook Viewer">`

_Possible Soultion_: 
Ensure the title attribute value presents the content and function of the image. For better accessibility, the alt attribute should be used when possible.
The code should look like this: `<img src="_static/nbviewer-thm-3.png" title="IPython Notebook Viewer" alt="IPython Notebook Viewer">`

4. ### Features
Alternative text is present for an image that is within a link. 
Including appropriate alternative text on an image within a link ensures that the function and purpose of the link and the content of the image is available to screen reader users or when images are unavailable.

_Current code_: `<img class="logo" src="_static/IPy_header.png" alt="Logo">`

_Possible solution_: Ensure that the alternative text presents the content of the image and/or the function of the link. If the full content and function of the link is presented in text within the link (an image and a text caption both within the same link, for example), then the image should generally be given empty/null alternative text (alt="") to avoid redundancy.

5. ### Structural elements

- *Missing heading levels* :  There are different heading levels that should be present, ranging from `<h1></h1>` to `<h3></h3>` and `<h4></h4>`. Headings facilitate page navigation for users of assistive technologies. They also provide semantic and visual meaning and structure to the document. First level headings should contain the most important heading(s) on the page (generally the document title).

_Possible solution_: Ensure that the text in question is truly a heading and that it is structured correctly in the page outline.

- *Unordered list*
_Current code_: 
`<ul class="simple">`
`<li>`
Very low contrast
`<a class="reference external" href="http://stackoverflow.com/questions/tagged/ipython">`
Stack Overflow
`</a>`
`</li>`
`<li>`
Very low contrast
`<a class="reference external" href="https://mail.python.org/mailman/listinfo/ipython-dev">`
Mailing list
`</a>`
`</li>`
`<li>`
Very low contrast
`<a class="reference external" href="https://github.com/ipython/ipython/issues">`
File a bug
`</a>`
`</li>`
`<li>`
Very low contrast
`<a class="reference external" href="http://www.reddit.com/r/IPython">`
Reddit
`</a>`
`</li>`
`</ul>`

_Possible solution_:
`<ol class="simple">`
`<li>`
Very low contrast
`<a class="reference external" href="http://stackoverflow.com/questions/tagged/ipython">`
Stack Overflow
`</a>`
`</li>`
`<li>`
Very low contrast
`<a class="reference external" href="https://mail.python.org/mailman/listinfo/ipython-dev">`
Mailing list
`</a>`
`</li>`
`<li>`
Very low contrast
`<a class="reference external" href="https://github.com/ipython/ipython/issues">`
File a bug
`</a>`
`</li>`
`<li>`
Very low contrast
`<a class="reference external" href="http://www.reddit.com/r/IPython">`
Reddit
`</a>`
`</li>`
`</ol>`

- *Inline frame*
This means an inline frame (<iframe>) is present.

_The problem_:
The content of an inline frame is read as if it were part of the page that contains it. The content of the iframe must be accessible. A title attribute value for the iframe will generally be read by a screen reader when the iframe is encountered.

_Possible solution_: 
Ensure that the content within the iframe is accessible. Optionally, a title attribute value can be added to provide a brief, advisory description of the iframe. In the opening tag of the iFrame element, a title attribute such as: `title=""`

- *Navigation*
A <nav> element or navigation landmark is present.
The navigation identifies a section of navigation links and can facilitate page semantics and navigation.

_Possible solution_:
Ensure the element defines page navigation. Multiple navigation elements on one page can be differentiated with ARIA labels.

- *Main content*
The `<main>` element or `role="main"` attribute identifies the main content for the page. This facilitate page semantics and navigation.

_Possible solution_:
Ensure the element surrounds and defines page main content.

- *Footer*
A `<footer>` element or contentinfo landmark is present.
Footers identify a footer for the page or a page section. It typically identifies authorship, related links, copyright date, or other footer content. Footers facilitate page semantics and navigation.

      _Possible solution_:
      Ensure the element surrounds and defines page or page section footer content.

- *Footer*
An aria-label or aria-labelledby attribute is present.
ARIA labels define accessible names to be read by screen readers for interface elements. They may be used when HTML associations (label, alternative text, etc.) are not sufficient.

_Possible solution_:
Ensure the aria-label value or element referenced by aria-labelledby provides a descriptive, accurate label. When possible, use standard HTML <`label>` or other markup to make the association.

### The HTML element to be changed

```html
<html xmlns="http://www.w3.org/1999/xhtml" style="margin-left: 380px;">

<img class="logo" src="_static/IPy_header.png" alt="Logo">

<img title="Packt Publishing donates a portion of the proceeds from this book to support IPython's development." width="160" height="200" id="sidebar-book-cover" style="margin-bottom: 10px;" src="_static/jupyter-notebook-for-all-II.jpg">

<img src="_static/jupytercon-logo.svg" width="180px" alt="JupyterCon 2017">

<h1>
Project
Very low contrast
<a class="headerlink" href="#project" title="Permalink to this headline">
</a>
</h1>

<ul class="simple">
</ul>

<iframe id="twitter-widget-0" scrolling="no" frameborder="0" allowtransparency="true" allowfullscreen="true" class="twitter-share-button twitter-share-button-rendered twitter-tweet-button" style="position: static; visibility: visible; width: 73px; height: 20px;" title="Twitter Tweet Button" src="https://platform.twitter.com/widgets/tweet_button.7dae38096d06923d683a2a807172322a.en.html#dnt=true&hashtags=ipython&id=twitter-widget-0&lang=en&original_referer=https%3A%2F%2Fipython.org%2Fproject.html&size=m&text=IPython%20-%20interactive%20computing.&time=1666445546920&type=share&url=http%3A%2F%2Fipython.org&via=ipythondev" data-url="http://ipython.org"></iframe>

<div class="sphinxsidebar" role="navigation" aria-label="main navigation">
```

### Your proposed HTML (or CSS), after change

```html
<html xmlns="http://www.w3.org/1999/xhtml" style="margin-left: 380px; lang="en" ">

<img class="logo" src="_static/IPy_header.png" alt="the logo of the wbsite">

<img title="Packt Publishing donates a portion of the proceeds from this book to support IPython's development." width="160" height="200" id="sidebar-book-cover" style="margin-bottom: 10px;" src="_static/jupyter-notebook-for-all-II.jpg" alt="a book about Jupyter">

<img src="_static/jupytercon-logo.svg" width="180px" alt="JupyterCon 2017 enclosed within a link">

<h2>
Project
Very low contrast
<a class="headerlink" href="#project" title="Permalink to this headline">
</a>
</h2>

<ol class="simple">
</ol>

<iframe title="twitter link" id="twitter-widget-0" scrolling="no" frameborder="0" allowtransparency="true" allowfullscreen="true" class="twitter-share-button twitter-share-button-rendered twitter-tweet-button" style="position: static; visibility: visible; width: 73px; height: 20px;" title="Twitter Tweet Button" src="https://platform.twitter.com/widgets/tweet_button.7dae38096d06923d683a2a807172322a.en.html#dnt=true&hashtags=ipython&id=twitter-widget-0&lang=en&original_referer=https%3A%2F%2Fipython.org%2Fproject.html&size=m&text=IPython%20-%20interactive%20computing.&time=1666445546920&type=share&url=http%3A%2F%2Fipython.org&via=ipythondev" data-url="http://ipython.org"></iframe>

<div class="sphinxsidebar" role="navigation" aria-label="a link for main navigation">
```

### More info

I believe that some of the issues above can also be solved or corrected with CSS too and not only HTML.

**Comment by minrk** [comment]
Thanks for the suggestions!

**Comment by sgibson91** [comment]
Thank you for your contribution! Since the contribution period is now over, we will close this issue.
